plugins/modules/katello_activation_key.py

```diff
@@ -145,88 +114,151 @@ def override_to_boolnone(override):
+    scope = {'organization_id': entity_dict['organization']['id']}
+    if not module.desired_absent:
+        if 'lifecycle_environment' in entity_dict:
+            entity_dict['lifecycle_environment'] = module.find_resource_by_name(
+                'lifecycle_environments', entity_dict['lifecycle_environment'], params=scope, thin=True)
 
-    if 'host_collections' in entity_dict:
-        entity_dict['host_collections'] = find_host_collections(module, entity_dict['host_collections'], entity_dict['organization'])
+        if 'content_view' in entity_dict:
+            entity_dict['content_view'] = module.find_resource_by_name('content_views', entity_dict['content_view'], params=scope, thin=True)
 
-    activation_key_entity = find_activation_key(module, name=entity_dict['name'], organization=entity_dict['organization'],
-                                                failsafe=True)
+    entity = module.find_resource_by_name('activation_keys', name=entity_dict['name'], params=scope, failsafe=True)
 
-    activation_key_dict = sanitize_entity_dict(entity_dict, name_map)
+    if module.state == 'copied':
+        new_entity = module.find_resource_by_name('activation_keys', name=entity_dict['new_name'], params=scope, failsafe=True)
+        if new_entity is not None:
+            module.warn("Activation Key '{}' already exists.".format(entity_dict['new_name']))
+            module.exit_json(changed=False)
 
-    try:
-        changed, activation_key_entity = naildown_entity(ActivationKey, activation_key_dict, activation_key_entity, state, module)
+    subscriptions = entity_dict.pop('subscriptions', None)
+    content_overrides = entity_dict.pop('content_overrides', None)
+    host_collections = entity_dict.pop('host_collections', None)
+    changed, activation_key = module.ensure_entity('activation_keys', entity_dict, entity, params=scope)
 
+    try:
         # only update subscriptions of newly created or updated AKs
         # copied keys inherit the subscriptions of the origin, so one would not have to specify them again
         # deleted keys don't need subscriptions anymore either
-        if state == 'present' or (state == 'present_with_defaults' and changed):
-            if 'subscriptions' in entity_dict:
-                subscriptions = entity_dict['subscriptions']
-                desired_subscription_ids = set(s.id for s in find_subscriptions(module, subscriptions, entity_dict['organization']))
-                current_subscriptions = [Subscription(**result)
-                                         for result in Subscription().search_normalize(activation_key_entity.subscriptions()['results'])]
-                current_subscription_ids = set(s.id for s in current_subscriptions)
+        if module.state == 'present' or (module.state == 'present_with_defaults' and changed):
+            ak_scope = {'activation_key_id': activation_key['id']}
+            # ak_scope.update(scope)  # See Hack above.
```

Comment:
which hack?

Reply:
the organization_id was `reqired=True` which renders it matter of luck to use the right api_url. And the wrong one delivers the wrong answers.
So it's the API-Hack i made above that still needs to be made nice.

Reply:
ah, ok, now it makes sense. even tho I don't fully understand why apypie would select the wrong url if we pass the org_id here. apypie bug?

Reply:
aah, after looking at the docs, there are multiple routes:

* GET /katello/api/organizations/:organization_id/subscriptions 
List organization subscriptions
* GET /katello/api/activation_keys/:activation_key_id/subscriptions 
List an activation key's subscriptions
* GET /katello/api/subscriptions 

You obviously want the second one. Makes sense.